src/datoso/repositories/dat.py

```diff
@@ -264,12 +264,15 @@ def get_next_block(self, data):
         parenthesis = 0
         start = 0
         end = 0
+        within_string = 0
```

Comment:
As this only takes two values better to make it boolean.
```suggestion
        within_string = False
```